tests/TestAppCpp/src/TestAppCpp.cpp

```diff
@@ -44,6 +51,136 @@ int main()
    return 0;
 }
 
+void TestParallel1(int argc, char** argv)
+{
+   string* simXMLStrings = NULL;
+   try
+   {
+      string runMode = "threads";
+
+      if (argc < 3)
+         throw "Missing arguments";
+      
+      int numberOfThreads = atoi(argv[2]);
+      if (numberOfThreads <= 0)
+         throw string("Number of threads must be >0. Passed value:") + argv[1];
+
+      if (argc > 3)
+      {
+         runMode = argv[3];
+      }
+
+      string filePath = argv[1];
+      auto simulationString = readFileIntoString(filePath);
+
+      cout << "Running '" << filePath.c_str() << "' (" << numberOfThreads << " runs)" << endl;
+
+      simXMLStrings = new string[numberOfThreads];
+      for (auto i = 0; i < numberOfThreads; i++)
+         simXMLStrings[i] = simulationString;
```

Comment:
Here I read a SimModel-XML file into string and create N copies of this string, so that every thread uses its own copy 
(guess it does not really matter, but just in case)